zwave_js_server/model/node.py

```diff
@@ -550,6 +530,48 @@ def highest_security_class(self) -> Optional[SecurityClass]:
             return None
         return SecurityClass(security_class)
 
+    @property
+    def is_controller_node(self) -> bool:
+        """Return whether the node is a controller node."""
+        return self.data["isControllerNode"]
+
+    @property
+    def keep_awake(self) -> bool:
+        """Return whether the node is set to keep awake."""
+        return self.data["keepAwake"]
+
+    def update_state(self, data: NodeDataType) -> None:
```

Comment:
In the other models we call it `update`.

Reply:
I was getting typing errors because the signature for this update method and the parent update method for endpoint were different. This was a lazy solution to fixing the problem

Reply:
That is a sign that our design is not optimal. Let's think about this a bit.

Reply:
maybe we shot ourselves in the foot by making Node a subclass of Endpoint. It does follow the zwave-js model to do so but from a practical standpoint I'm not sure we get a lot of benefits from modeling it this way

Reply:
so to follow up on this, there are two options in my mind:
1. We match the signature of the update function on both classes. This would mean passing the values separately from the rest of the node state data into the update function, which is fine but a little ugly in my opinion.
2. We break the relationship between `Endpoint` and `Node` and with them being independent, there is no problem

Neither solution is optimal in my opinion, but I am sort of leaning towards 1 because it's less of a model break and more of a convention break. Thoughts?

Reply:
Actually 1 would create issues because in Endpoint we pass `Value`/`ConfigurationValue` whereas in Node we get `ValueDataType`. We can't pass `Value`/`ConfigurationValue` into node because we want the node to update the value data accordingly, and we can't pass `ValueDataType` into `Endpoint` because then we'd be creating completely new `Value` instances and we should have the instances be the same on both `Endpoint` and `Node`.

OK so adding two more options (assuming we want to pursue 1):
1. We update the typing of the update method to be `List[Union[ValueDataType, Value, ConfigurationValue]]` and assert the types in each class's update method
2. We always pass ValueDataType to both `Endpoint` and `Node`. Since `Node` is always updated first, the `Endpoint` can retrieve the `Value` from the node using the `ValueDataType` data.

1 seems cleaner to me but it may not address your concerns with the model?

Reply:
The typing will not work no matter what I try for option 1 (trying to keep the subclass relationship). I think the best thing to do at this point is to break the relationship and make them independent classes.